hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/IngresDialect.java

```diff
@@ -279,10 +279,16 @@ public void initializeFunctionRegistry(QueryEngine queryEngine) {
 		CommonFunctionFactory.format_dateFormat( queryEngine );
 		CommonFunctionFactory.dateTrunc( queryEngine );
 		CommonFunctionFactory.bitLength_pattern( queryEngine, "octet_length(hex(?1))*4" );
-		CommonFunctionFactory.locate_positionSubstring( queryEngine );
 
 		final BasicType<Integer> integerType = queryEngine.getTypeConfiguration().getBasicTypeRegistry()
 				.resolve( StandardBasicTypes.INTEGER );
+		queryEngine.getSqmFunctionRegistry().registerBinaryTernaryPattern(
+				"locate",
+				integerType,
+				"position(?1 in ?2)",
+				"(position(?1 in substring(?2 from ?3))+(?3)-1)",
+				STRING, STRING, INTEGER
+		).setArgumentListSignature("(pattern, string[, start])");
```

Comment:
IMO this change was correct. According to the docs(https://docs.actian.com/ingres/10.2/index.html#page/SQLRef/String_Functions.htm#ww417057) the position function returns a 1 based index, just like the JPA spec requires it.